src/main/kotlin/org/cascadebot/cascadebot/commands/moderation/MuteChannelSetupSubCommand.kt

```diff
@@ -0,0 +1,120 @@
+import net.dv8tion.jda.api.entities.Member
+import net.dv8tion.jda.api.entities.TextChannel
+import net.dv8tion.jda.api.entities.VoiceChannel
+import net.dv8tion.jda.api.exceptions.InsufficientPermissionException
+import org.cascadebot.cascadebot.commandmeta.CommandContext
+import org.cascadebot.cascadebot.commandmeta.SubCommand
+import org.cascadebot.cascadebot.messaging.MessageType
+import org.cascadebot.cascadebot.messaging.Messaging
+import org.cascadebot.cascadebot.permissions.CascadePermission
+import org.cascadebot.cascadebot.utils.ConfirmUtils
+import org.cascadebot.cascadebot.utils.getMutedRole
+import java.util.concurrent.CompletableFuture
+
+class MuteChannelSetupSubCommand : SubCommand() {
+
+    override fun onCommand(sender: Member, context: CommandContext) {
+        if (!context.selfMember.hasPermission(Permission.MANAGE_CHANNEL)) {
+            context.uiMessaging.sendBotDiscordPermError(Permission.MANAGE_CHANNEL)
+            return
+        } else if (!context.selfMember.hasPermission(Permission.MANAGE_PERMISSIONS)) {
+            context.uiMessaging.sendBotDiscordPermError(Permission.MANAGE_PERMISSIONS)
+            return
+        }
+
+        if (ConfirmUtils.hasConfirmedAction("mute.channelsetup", context.user.idLong)) {
+            ConfirmUtils.completeAction("mute.channelsetup", context.user.idLong)
+            return
+        }
+
+        ConfirmUtils.confirmAction(
+                context.user.idLong,
+                "mute.channelsetup",
+                context.channel,
+                MessageType.WARNING,
+                context.i18n("commands.mute.channelsetup.warning"),
+                true,
+                object : ConfirmUtils.ConfirmRunnable() {
+                    override fun execute() {
+                        channelSetup(context)
+                    }
+                }
+        )
+    }
+
+    override fun command(): String = "channelsetup"
+
+    override fun parent(): String = "mute"
+
+    override fun permission(): CascadePermission = CascadePermission.of("mute.channelsetup", false)
+
+    private fun channelSetup(context: CommandContext) {
+        val mutedRole = context.guild.getMutedRole()
+        val success = mutableListOf<GuildChannel>()
+        val failed = mutableListOf<GuildChannel>()
+
+        val futures: MutableList<CompletableFuture<Void>> = mutableListOf()
+
+        context.guild.channels.forEach {
+            try {
+                futures.add(when (it) {
+                    is TextChannel -> it.manager.putPermissionOverride(mutedRole, setOf(), setOf(Permission.MESSAGE_WRITE)).submit()
+                    is VoiceChannel -> it.manager.putPermissionOverride(mutedRole, setOf(), setOf(Permission.VOICE_SPEAK)).submit()
+                    is Category -> {
+                        it.manager.putPermissionOverride(mutedRole, setOf(), setOf(Permission.MESSAGE_WRITE, Permission.VOICE_SPEAK)).submit()
+                    }
+
+                    else -> return@forEach
```

Comment:
Have you tested if this respects channels synced with the category in terms of permissions? As in this doesn't unsync it due to setting the individual channel permissions. If it does unsync it we'll need to add check so we don't unsync it.

Reply:
Funnily enough, I found out the "channel sync" is actually handled in an interesting way. A channel is considered synced if all of it's permissions are the same as the category.

Reply:
All so the client must do the syncing then. Alright then this is fine.